crawler/helpers/cherrypicked_samples.py

```diff
@@ -0,0 +1,199 @@
+    dataframe only contains those samples that have been cherrypicked (those that have an entry
+    for the relevant event type in the event warehouse)
+
+    Args:
+        root_sample_ids (List[str]): [description]
+        plate_barcodes (List[str]): [description]
+        chunk_size (int, optional): [description]. Defaults to 50000.
+
+    Returns:
+        DataFrame: [description]
+    """
+    logger.debug("Getting cherry-picked samples from MLWH")
+
+    return build_data_frame_from_database_queries_aggregation(
+        database_connection_uri(config),
+        SQL_MLWH_GET_CP_SAMPLES,
+        params_for_cherrypicked_samples_query(root_sample_ids, plate_barcodes, chunk_size),
+    )
+
+
+def get_cherrypicked_samples_by_date(
+    config: Config,
+    root_sample_ids: List[str],
+    plate_barcodes: List[str],
+    start_date: str,
+    end_date: str,
+    chunk_size: int = 50000,
+) -> Optional[DataFrame]:
+    """Find which samples have been cherrypicked between defined dates using MLWH & Events warehouse.
+    Returns dataframe with 4 columns: those needed to uniquely identify the sample resulting
+    dataframe only contains those samples that have been cherrypicked (those that have an entry
+    for the relevant event type in the event warehouse)
+
+    Args:
+        config (Config): application config specifying database details
+        root_sample_ids (List[str]): [description]
+        plate_barcodes (List[str]): [description]
+        start_date (str): lower limit on creation date
+        end_date (str): upper limit on creation date
+        chunk_size (int, optional): [description]. Defaults to 50000.
+
+    Returns:
+        DataFrame: [description]
+    """
+
+    logger.debug("Getting cherry-picked samples from MLWH")
+
+    return build_data_frame_from_database_queries_aggregation(
+        database_connection_uri(config),
+        SQL_MLWH_GET_CP_SAMPLES_BY_DATE,
+        params_for_cherrypicked_samples_by_date_query(
+            root_sample_ids, plate_barcodes, start_date, end_date, chunk_size
+        ),
+    )
+
+
+def remove_cherrypicked_samples(samples: List[SampleDoc], cherry_picked_samples: List[List[str]]) -> List[SampleDoc]:
+    """Remove samples that have been cherry-picked. We need to check on (root sample id, plate barcode) combo rather
+    than just root sample id. As multiple samples can exist with the same root sample id, with the potential for one
+    being cherry-picked, and one not.
+
+    Args:
+        samples (List[Sample]): List of samples in the shape of mongo documents
+        cherry_picked_samples (List[List[str]]): 2 dimensional list of cherry-picked samples with root sample id and
+        plate barcodes for each.
+
+    Returns:
+        List[Sample]: The original list of samples minus the cherry-picked samples.
+    """
+    cherry_picked_sets = [{cp_sample[0], cp_sample[1]} for cp_sample in cherry_picked_samples]
+    return list(
+        filter(
```

Comment:
First store the filter's results before converting to list to make it more readable?